core/src/main/kotlin/org/evomaster/core/search/gene/ChoiceGene.kt

```diff
@@ -0,0 +1,105 @@
+package org.evomaster.core.search.gene
+
+import org.evomaster.core.logging.LoggingUtil
+import org.evomaster.core.output.OutputFormat
+import org.evomaster.core.search.service.AdaptiveParameterControl
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneSelectionStrategy
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+
+class ChoiceGene<T>(name: String,
+                 private val choices: List<T>,
+                 activeChoice: Int = 0
+
+) : CompositeFixedGene(name, choices) where T : Gene {
+
+    companion object {
+        private val log: Logger = LoggerFactory.getLogger(ChoiceGene::class.java)
+    }
+
+    var activeChoice: Int = activeChoice
+        private set
+
+    init {
+        if (activeChoice < 0 || activeChoice >= choices.size) {
```

Comment:
add explicit check that `choices` is not empty

Reply:
ok. But still the constructor of CompositeFixedGene fails if the children list is empty.